docs: restructure docs for new hosting

owenrumney opened the pull request:
## Description

This PR restructures the docs a little to support the new hosting in
CloudFlare and breaking away the home page management to marketting
while retaining a workable docs site.

Key changes -

- rename docs/docs to docs/guide as a path
- rename Docs menu to User Guide
- update the index in docs/guide to reflect the new name of User Guide
- remove the assets that were for partner and home page
- remove the partner page from the nav
- update the links to point to guide rather than docs where applicable
- update the mage file to remove the redunant CSS target
- update the links in the the mage docs for specialist references files

>[!Note]
>Home and Partners is rewritten during the deployment to CloudFlare from the rather empty `docs/index.md` to the root of the Marketting site

## Related issues
- Close #9798

## Checklist
- [x] I've read the [guidelines for contributing](https://trivy.dev/docs/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://trivy.dev/docs/latest/community/contribute/pr/#title) in the PR title.

knqyf263 commented:
We probably need to update `mkdocs.yml` as `Partners` is still shown and returns 404.

<img width="816" height="237" alt="image" src="https://github.com/user-attachments/assets/1d04b690-ce22-40ff-a83a-bc5a742875d8" />

owenrumney commented:
> > Home and Partners is rewritten during the deployment to CloudFlare from the rather empty docs/index.md to the root of the Marketting site
> 
> Oh, sorry I missed it. It seems more intuitive to keep MkDocs running on its own instead of returning a 404 for partners, and it might also be less confusing for external contributors (and even maintainers like me 😄 ), but I’m fine either way.

Don't worry, you didn't miss it, I added it to make it more clear :-D

I think removing it is probably best for the developer experience - if Itay wants it in the docs, it can be added at the trivy-www level.

I've updated the PR.
